src/a-runtime-console/kubernetes/ui/pipeline/input-action-dialog/input-action-dialog.component.ts

```diff
@@ -84,4 +88,73 @@ export class InputActionDialog {
   close() {
     this.modal.close();
   }
+
+  checkIfJenkinsIsUpAndProceedURL() {
+    let notification;
+    this.callJenkins()
+      .subscribe((response) => {
+        switch (response.status) {
+          case 200:
+            notification = {
+              message: `Got ${response.status}, Jenkins is up and running.`,
+              type: NotificationType.SUCCESS
+            };
+            this.notifications.message(notification as Notification);
+            this.invokeUrl(this.inputAction.proceedUrl);
+            return;
+          case 307:
+            notification = {
+              message: `Got ${response.status}, Connecting to Jenkins.`,
+              type: NotificationType.INFO
+            };
+            break;
+          case 503:
+            notification = {
+              message: `Got ${response.status}, Cluster resources capacity is full. Waiting along.`,
+              type: NotificationType.INFO
+            };
+            break;
+          case 504:
+            notification = {
+              message: `Got ${response.status}, Request timed out. Trying again.`,
+              type: NotificationType.WARNING
+            };
+            break;
+          case 202:
+            notification = {
+              message: `Got ${response.status}, Jenkins is currently idled. Please wait while it starts.`,
+              type: NotificationType.WARNING
+            };
+            break;
+          default:
+            notification = {
+              message: `Got status ${response.status}`,
+              type: NotificationType.DANGER
+            };
+            this.notifications.message(notification as Notification);
+            this.close();
```

Comment:
This is within an asynchronous block called from `checkIfJenkinsIsUpAndProceedURL()`. This function always calls `this.close()` before returning (line 141) so I don't think the default case needs to also call `this.close()`. Thoughts?

Reply:
@jiekang I agree. Removing `this.close()` from default case. I added it earlier thinking that if default case returns then the modal might not be closed but that's not the case since switch is under .subscribe() and `this.close()` in the end always gets called.